packages/dev/core/src/Engines/nativeEngine.ts

```diff
@@ -1635,9 +1635,14 @@ export class NativeEngine extends Engine {
         }
 
         if (!!texture && !!texture._hardwareTexture) {
+            const context = canvas.getContext();
+            context.flush();

```

Comment:
Is it really ok to do this synchronously right here? Couldn't queued commands still alter the state of the canvas/context? Should the flush itself be queued, even if the flush has to be a blocking operation on the native side?

Reply:
As long as the flush (and nanovg rendering) happens before the copy, it will be fine. The queue is managed by NativeEngine so, the context/canvas won't be aware of the flush command in the queue.